joblib/numpy_pickle.py

```diff
@@ -216,9 +216,8 @@ def __init__(self, filename, compress=0, cache_size=10):
             self.file = BytesIO()
         # Count the number of npy files that we have created:
         self._npy_counter = 0
-        highest_python_2_3_compatible_protocol = 2
         Pickler.__init__(self, self.file,
-                         protocol=highest_python_2_3_compatible_protocol)
+                         protocol=pickle.HIGHEST_PROTOCOL)
```

Comment:
I think we should use pickle protocol = 2 for Python 2 and pickle protocol = 3 (default protocol) for Python 3.